libpod/container_internal_linux.go

```diff
@@ -698,21 +698,37 @@ func (c *Container) makeBindMounts() error {
 			// If it doesn't, don't copy them
 			resolvPath, exists := bindMounts["/etc/resolv.conf"]
 			if exists {
-
 				c.state.BindMounts["/etc/resolv.conf"] = resolvPath
 			}
+
+			// check if dependency container has an /etc/hosts file
 			hostsPath, exists := bindMounts["/etc/hosts"]
-			if exists {
-				c.state.BindMounts["/etc/hosts"] = hostsPath
+			if !exists {
+				return errors.Errorf("error finding hosts file of dependency container %s for container %s", depCtr.ID(), c.ID())
+			}
+
+			depCtr.lock.Lock()
+			// generate a hosts file for the dependency container,
+			// based on either its old hosts file, or the default,
+			// and add the relevant information from the new container (hosts and IP)
+			hostsPath, err = depCtr.generateHosts(hostsPath, c)
```

Comment:
Hmm. I think this would replace the file, which might now work. Might be safer to open it for append and write the new entries?

Reply:
Not work, even

Reply:
right now it's actually taking the old file, appending to it, and writing it all back (replacing it). I had it this way to have the actual generate code (make a new hosts file) and the append (when jumping into a netNS) the same function. Should I seperate?

Reply:
`writeStringToRundir()` removes the file in question (if present) and recreates it, which is my concern. I think we need a separate append version, yeah.

Reply:
Fixed!